eva-accession-release/src/main/java/uk/ac/ebi/eva/accession/release/io/AccessionedVariantMongoReader.java

```diff
@@ -60,35 +59,35 @@
 
     private static final String DBSNP_CLUSTERED_VARIANT_ENTITY = "dbsnpClusteredVariantEntity";
 
-    private static final String DBSNP_SUBMITTED_VARIANT_ENTITY = "dbsnpSubmittedVariantEntity";
+    protected static final String DBSNP_SUBMITTED_VARIANT_ENTITY = "dbsnpSubmittedVariantEntity";
 
-    private static final String ACCESSION_FIELD = "accession";
+    protected static final String ACCESSION_FIELD = "accession";
 
     private static final String REFERENCE_ASSEMBLY_FIELD = "asm";
 
-    private static final String STUDY_FIELD = "study";
+    protected static final String STUDY_FIELD = "study";
 
     private static final String CONTIG_FIELD = "contig";
```

Comment:
Some of these are private but also used in the derived class. Maybe we can just make all of them protected, they aren't going to pollute the context of the class a lot more.

Another point is that I don't really see `MergedVariantMongoReader`'s implementation as really derived from `AccessionedVariantMongoReader`. I think using a [Template Method](https://en.wikipedia.org/wiki/Template_method_pattern) that implements `ItemStreamReader`, then making the 2 concrete readers extend it, would improve the code structure.

Reply:
the part about the aggregation fits the template pattern, but the method `buildVariantSourceEntry` is used in both implementations, and the Merged one only adds an extra variable, so the template pattern doesn't fit very well for that part. Can you think of a better structure (using template pattern) than:
```
///////////// Abstract parent class
protected VariantSourceEntry buildVariantSourceEntry( ... ) {
    new VariantSourceEntry ...
    sourceEntry.addAttribute ...
    // more attributes
    implUpdateVariantSourceEntry(sourceEntry) // overrided
    return sourceEntry;
}

////////////// Implementation class
add variants() {
    // complex method with lots of variables (only way to refactor is to create a class similar to Variant to hold them, which is pointless, or make them fields, and change them for each iteration)
    for ( ... ) {
        buildVariantSourceEntry( ..., **extraParameterMergedInto**)
    }
}
buildVariantSourceEntry( ... , **extraParameterMergedInto**) {
    this.extraParameterMergedInto = extraParameterMergedInto // only way I can think of. ugly
    super.buildVariantSourceEntry(...)
}
implUpdateVariantSourceEntry(sourceEntry){
    sourceEntry.addAttribute(extraParameterMergedInto) // where do we get this parameter? a field?
}
```
or do you prefer mixing both patterns and do the template pattern for the aggregation, and leave it as it is now (simple parent reuse) for the `buildVariantSourceEntry` part?

Reply:
also, the fields that I left private, are not the same. "asm", "study", "contig", "start", "type"; in one class they refer to DbsnpClusteredVariantEntity fields, and the other to the equivalent fields in DbsnpClusteredVariantInactiveEntity. should I still reuse them?

Reply:
> in one class they refer to DbsnpClusteredVariantEntity fields, and the other to the equivalent fields in DbsnpClusteredVariantInactiveEntity. should I still reuse them?

The entities have the same internal structure, but one is simply flagged as inactive. I think we should reuse them.